### PR TITLE
flamegraph: 2017-07-01 -> 2019-02-16

### PR DESCRIPTION
**Pull request by cko**
###### Motivation for this change
Updating flamegraph

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
  - I tested `stackcollapse-perf.pl` and `flamegraph.pl`
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build flamegraph

File: pkgs/development/tools/flamegraph/default.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
 { stdenv, fetchFromGitHub, perl }:
 
 stdenv.mkDerivation {
-  name = "FlameGraph-2017-07-01";
+  name = "FlameGraph-2019-02-16";
```
Comment (c0bw3b) [inline pkgs/development/tools/flamegraph/default.nix]:
```suggestion
  pname = "FlameGraph";
  version = "2019-02-16";
```

File: pkgs/development/tools/flamegraph/default.nix [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
   src = fetchFromGitHub {
     owner = "brendangregg";
     repo = "FlameGraph";
```
Comment (c0bw3b) [inline pkgs/development/tools/flamegraph/default.nix]:
```suggestion
    repo = pname;
```

Comment (c0bw3b) [inline pkgs/development/tools/flamegraph/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/tools/flamegraph/default.nix [hunk 3]
```diff
@@ -21,7 +21,7 @@ stdenv.mkDerivation {
 
   meta = with stdenv.lib; {
     license = licenses.cddl;
```
Comment (c0bw3b) [inline pkgs/development/tools/flamegraph/default.nix]:
```suggestion
    license = with licenses [ asl20 cddl gpl2Plus ];
```
Looking at some of the scripts' headers it seems they come under one of these licenses.

File: pkgs/development/tools/flamegraph/default.nix [hunk 4]
```diff
@@ -1,13 +1,13 @@
 { stdenv, fetchFromGitHub, perl }:
 
 stdenv.mkDerivation {
```
Comment (c0bw3b) [inline pkgs/development/tools/flamegraph/default.nix]:
```suggestion
stdenv.mkDerivation rec {
```
(in order to be able to set `repo = pname` further down)

**Comment by c0bw3b** [comment]
When overriding an entire phase, it's best practice to call pre- and post- hooks in case they are needed later on.

```nix
installPhase = ''
  runHook preInstall
  [...]
  runHook postInstall
'';
```

**Comment by cko** [comment]
@c0bw3b I implemented the proposed changes and updated the PR.

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build flamegraph

**Review by c0bw3b — APPROVED** [review]

**Comment by c0bw3b** [comment]
Thanks @cko

**Pull request merged**